### PR TITLE
Issue 1015/campaign activities tab

### LINKED ISSUE #1015: Basic "Campaign activities" tab
## Description
A user should be able to find surveys, events and other activities of a campaigning in a single place, a tab on the campaign's page.

## Relevant Job Stories
* When my survey has been live for a while, I want to be able to easily find it, so that I can go look at the responses
* When I’m running a survey, I want to be able to keep it alongside related content such as call assignments and events, so that I can find them all in a single place

## Prerequisites
None

## Requirements
- [x] The campaign page has an "Activities" tab
- [ ] The list includes surveys, tasks and call assignments
- [x] The list is sorted chronologically

## Possible implementations
None

## Design specifications

Image

## Open questions
* What date is used for sorting? Surveys, tasks and call assignments (can) have start- and end dates.

### PR DESCRIPTION
**Pull request by ziggabyte**
## Description
This PR adds an Activities tab to each project, that displays the ActivitiyList. It also refactors to make a NoActivities component. 

## Screenshots
bild

## Changes
* Adds Activities tab to projects
* Creates NoActivities component.

## Related issues
Resolves #1015

### CODE REVIEW

File: src/pages/organize/[orgId]/campaigns/[campId]/activities/index.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+const CampaignActivitiesPage: PageWithLayout<CampaignActivitiesPageProps> = ({
+  campId,
+  orgId,
+}) => {
+  const messages = useMessages(messageIds);
+  const onServer = useServerSide();
+  const model = useModel(
+    (env) => new CampaignActivitiesModel(env, parseInt(orgId))
+  );
+  const activities = model
```
Comment (richardolsson) [inline src/pages/organize/[orgId]/campaigns/[campId]/activities/index.tsx:46]: Maybe this filtering could be moved _into_ the model? I'll let you decide!

**Review by richardolsson — APPROVED** [review]
I have a little piece of feedback but in general this looks great! I've only looked at the code, not verified that it works.

File: src/pages/organize/[orgId]/campaigns/[campId]/activities/index.tsx (hunk 1 above)
Comment (ziggabyte) [reply src/pages/organize/[orgId]/campaigns/[campId]/activities/index.tsx:46]: so make two different methods for the different types of filtering (here for campaign id, and standalones on the pther page)?

Comment (richardolsson) [reply src/pages/organize/[orgId]/campaigns/[campId]/activities/index.tsx:46]: Yes, that's what I was envisioning, but I haven't thought it through so I don't know how much of an improvement that would be. You decide! 😊

Comment (richardolsson) [reply src/pages/organize/[orgId]/campaigns/[campId]/activities/index.tsx:46]: Yes, that's what I was envisioning, but I haven't thought it through so I don't know how much of an improvement that would be. You decide! 😊

Comment (ziggabyte) [reply src/pages/organize/[orgId]/campaigns/[campId]/activities/index.tsx:46]: will test and see!

File: src/features/campaigns/models/CampaignAcitivitiesModel.ts
Comment (rebecarubio) [inline src/features/campaigns/models/CampaignAcitivitiesModel.ts:1]: I think there is a typo in the name of the file 🧐

Comment (ziggabyte) [reply src/features/campaigns/models/CampaignAcitivitiesModel.ts:1]: THank YOU!

File: src/features/campaigns/models/CampaignActivitiesModel.ts [hunk 2]
```diff
@@ -33,6 +33,14 @@ export default class CampaignActivitiesModel extends ModelBase {
     this._tasksRepo = new TasksRepo(env);
   }
 
+  getCampaignActivities(campId: number): IFuture<CampaignAcitivity[]> {
+    const activities = this.getCurrentActivities().data;
+    const filtered = activities?.filter(
+      (activity) => activity.campaign?.id === campId
```
Comment (richardolsson) [inline src/features/campaigns/models/CampaignActivitiesModel.ts:39]: Good catch!

**Review by richardolsson — APPROVED** [review]
I feel like this is ready to merge!

**Pull request merged**